tests/test_cmdline.py

```diff
@@ -399,166 +396,154 @@ def test_list_deps__pick_multiple_listed_files__prints_all_dependencies(
+            "These imports appear to be undeclared dependencies:",
+            "- 'requests'",
+            "",
+            "These dependencies appear to be unused (i.e. not imported):",
+            "- 'pandas'",
+            "",
+            VERBOSE_PROMPT,
+        ],
+        expect_logs=[
+            "INFO:fawltydeps.extract_imports:Parsing Python files under {path}",
+            "INFO:fawltydeps.packages:'pandas' was not resolved."
+            " Assuming it can be imported as 'pandas'.",
+        ],
+        expect_returncode=3,  # undeclared is more important than unused
+    ),
+    ProjectTestVector(
+        id="simple_project__summary_report_with_quiet_logging",
+        options=["--check", "--detailed", "--code={path}", "--deps={path}"],
+        imports=["requests"],
+        declares=["pandas"],
+        expect_output=[
+            "These imports appear to be undeclared dependencies:",
+            "- 'requests' imported at:",
+            "    {path}/code.py:1",
+            "",
+            "These dependencies appear to be unused (i.e. not imported):",
+            "- 'pandas' declared in:",
+            "    {path}/requirements.txt",
+        ],
+        expect_returncode=3,  # undeclared is more important than unused
+    ),
+]
 
 
-def test_check__simple_project__detailed_report_with_quiet_logging(
+@pytest.mark.parametrize(
+    "vector", [pytest.param(v, id=v.id) for v in project_tests_samples]
+)
+def test_check_undeclared_and_unused(
+    vector,
     project_with_code_and_requirements_txt,
 ):
     tmp_path = project_with_code_and_requirements_txt(
-        imports=["requests"],
-        declares=["pandas"],
+        imports=vector.imports,
+        declares=vector.declares,
     )
-
-    expect = [
-        "These imports appear to be undeclared dependencies:",
-        "- 'requests' imported at:",
-        f"    {tmp_path / 'code.py'}:1",
-        "",
-        "These dependencies appear to be unused (i.e. not imported):",
-        "- 'pandas' declared in:",
-        f"    {tmp_path / 'requirements.txt'}",
-    ]
-    expect_logs = ""
-    output, errors, returncode = run_fawltydeps(
-        "--check", "--detailed", f"--code={tmp_path}", f"--deps={tmp_path}"
+    output, logs, returncode = run_fawltydeps(
+        *[option.format(path=tmp_path) for option in vector.options]
     )
-    assert output.splitlines() == expect
-    assert errors == expect_logs
-    assert returncode == 3  # undeclared is more important than unused
+    # Order of output is determined, as we use alphabetical ordering.
+    assert output.splitlines() == [
+        v.format(path=tmp_path) for v in vector.expect_output
+    ]
+    # We do set comparison here, as the evaluation order of
+    # input is not determined. In the process of evaluation,
+    # we use sets (like in IdentityMapping fallback).
+    assert {l for l in logs.split("\n") if l != ""} == {
+        v.format(path=tmp_path) for v in vector.expect_logs
+    }
```

Comment:
It might be relevant that the test that is [failing the CI](https://github.com/tweag/FawltyDeps/actions/runs/4615271918/jobs/8158990052) in the commit before the last one is `tests/test_cmdline.py::test_check_undeclared_and_unused[simple_project_with_extra_deps__reports_unused]`, which has a different log assertion from the other similar tests:
```python
assert_unordered_equivalence(errors.splitlines(), expect_logs)
```
Compared to:
```
assert errors == "\n".join(expect_logs)
```

So this unordered set of logs might have been a concern before and had led to this difference.

Do you think this will still work if you change the set comparison you provide above with `assert_unordered_equivalence`?

Reply:
Set comparison is not the only reason why this like looks like this. There is also filtering of the empty lines on the LHS  and formatting of the expected output string.